src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,67 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    /**
+     * Return the list of machines.
+     * In the implementation of this method please use your MachineProducer implementations
+     * See that 'Class type' is not parametrized.
+     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
+     *
+     * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
+     * @return the list of machines
+     */
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> machineProducer;
+        if (Bulldozer.class.equals(type)) {
+            machineProducer = new BulldozerProducer();
+        } else if (Excavator.class.equals(type)) {
+            machineProducer = new ExcavatorProducer();
+        } else if (Truck.class.equals(type)) {
+            machineProducer = new TruckProducer();
+        } else {
+            return new ArrayList<>();
+        }
+        return (List<Machine>)machineProducer.get();
```

Comment:
```suggestion
        if (Bulldozer.class.equals(type)) {
            return buldozzerProducer.get();
        } 
        if (Excavator.class.equals(type)) {
            ....
        }
        ....
        return return new ArrayList<>();
```

lets make MachineProducers as class level variables, not to create new objects every time we call getAll method?